### PR TITLE
tiled: Support tile flipping flags

### PR DESCRIPTION
**Pull request by septum**
Based from this [documentation reference](https://doc.mapeditor.org/en/stable/reference/tmx-map-format/#tile-flipping).

Beware: code may be ugly and unidiomatic.

### CODE REVIEW

File: tiled/src/lib.rs [hunk 1]
```diff
@@ -92,22 +101,48 @@ impl Map {
...
+                    source: Some(Rect::new(
+                        spr_rect.x - 1.0,
+                        spr_rect.y - 1.0,
+                        spr_rect.w + 2.0,
+                        spr_rect.h + 2.0,
+                    )),
+                    flip_x: flags.horizontally,
+                    flip_y: flags.vertically,
+                    rotation: if flags.antidiagonally {
+                        -90. * f32::consts::PI / 180.
```
Comment (vdrn) [inline tiled/src/lib.rs:121]: The issue is here. We need to multiply by 90 instead of -90.

**Review by vdrn — CHANGES_REQUESTED** [review]
This feature requires fixing rotation code. At line 121@tiled/src/lib.rs we should multiply by `90` instead of `-90`.

**Comment by vdrn** [comment]
> This feature requires fixing rotation code. At line 121@tiled/src/lib.rs we should multiply by `90` instead of `-90`.

Please disregard my previous comment.


Currrently this works for me, though I'm sure there must be  a simpler method to do this:
```
flip_x: (flags.horizontally
    && (!flags.antidiagonally || (flags.vertically && flags.antidiagonally)))
    || (!flags.horizontally && flags.antidiagonally && !flags.vertically),  
    
flip_y: flags.vertically && !flags.antidiagonally && !flags.horizontally,  

rotation: if flags.antidiagonally {  
    if !flags.horizontally {  
        -f32::consts::PI / 2.0  
    } else {  
        f32::consts::PI / 2.0  
    }  
} else {
    0.
}
```

**Comment by septum** [comment]
I'll close this now.

**Pull request closed without merging**